doc/rst/developer/bestPractices/GeneratedCode.rst

```diff
@@ -176,127 +176,33 @@ then call ``sdirs`` from the gdb prompt (do not use `~` or `$CHPL_HOME`), e.g.:
-*the same* version of the generated code.
-
-
-* Keep track of how your runtime is presently built (see above).
-
-
-* When compiling, use ``--savec`` but not ``-g`` (I think), ``--ccflags``, or ``--ldflags``,
-  for example:
-
-      ``--fast --savec DIR --c-line-numbers``
-
-* Option A: run ``make -f DIR/Makefile`` then adjust the compilation
-  commands being issued. (You might even be able to redirect the
-  compilation to different runtime/lib directories.)
-
-
-* Option B is to replace `DIR/Makefile`
-  with the following (change ``a.out`` to your preferred executable name):
-
-.. code-block :: bash
-
-
-  ifneq ($(DB),)
-  EXTR_FLAGS += -g
-  EXTR_sfx += .db
-  endif
-
-  ifneq ($(GP),)
-  EXTR_FLAGS += -pg
-  EXTR_sfx += .gp
-  endif
-
-  ifneq ($(GC),)
-  EXTR_FLAGS += -fprofile-arcs -ftest-coverage
-  EXTR_sfx += .gc
-  endif
-
-  ifneq ($(DB),)
-  # don't want OPT_CFLAGS
-  COMP_GEN_CFLAGS = $(EXTR_FLAGS) $(WARN_GEN_CFLAGS)               $(NO_IEEE_FLOAT_GEN_CFLAGS)
-  else
-  COMP_GEN_CFLAGS = $(EXTR_FLAGS) $(WARN_GEN_CFLAGS) $(OPT_CFLAGS) $(NO_IEEE_FLOAT_GEN_CFLAGS)
-  endif
-
-  COMP_GEN_LFLAGS = $(EXTR_FLAGS) 
-  BINNAME = a.out$(EXTR_sfx)
-  TMPDIRNAME := $(dir $(lastword $(MAKEFILE_LIST)))
-  TMPBINNAME = $(TMPDIRNAME)a.out.tmp
-  CHAPEL_ROOT = $(CHPL_HOME)
-  TAGS_COMMAND = -@which $(CHPL_TAGS_UTIL) > /dev/null 2>&1 && test -f $(CHAPEL_ROOT)/runtime/$(CHPL_TAGS_FILE) && cd $(TMPDIRNAME) && cp $(CHAPEL_ROOT)/runtime/$(CHPL_TAGS_FILE) . && $(CHPL_TAGS_UTIL) $(CHPL_TAGS_FLAGS) $(CHPL_TAGS_APPEND_FLAG) *.c *.h
-
-  CHPLSRC = $(TMPDIRNAME)_main.c 
-  LIBS =
-
-  include $(CHAPEL_ROOT)/runtime/etc/Makefile.include
-
-
-Do all compilations in DIR (e.g. the directory with the generated code).
-
-::
-
-      make DB=1   # generates 'a.out.db' for debugging
-      make GP=1   # generates 'a.out.gp' for gprof
-      make GC=1   # generates 'a.out.gc' for gcov
-
-For ``gcov``, you may need to rename a couple of files by hand:
-
-::
-
-      rm *.c.gcov *.gcda *.gcno
-      make GC=1
-      ./a.out.gc <whatever options you have for a.out>
-      mv a.out{.tmp,}.gcda
-      mv a.out{.tmp,}.gcno
-      gcov a.out.gc
-
-
-Profiling only parts of the runtime
------------------------------------
-
-Do you want ``gprof/gcov/...`` to look only at certain parts of the runtime,
-to reduce profiling overhead, making the results more true to reality?
-
-One way to do so is to transplant those parts from the runtime into
-the generated code, patching everything involved to satisfaction
-(e.g. so that those parts compile and the generated code refers to
-them instead of the original runtime). This can easily be laborious,
-but you get good control over what's going on.
-
-Be sure NOT to re-run the Chapel compiler.
+  This produces a `gprof-enabled` executable.
 
+  The use of ``--fast`` and ``--savec`` are not strictly necessary,
+  but will improve exeution time and give you access to the generated
+  C code (in the named directory), respectively.
```

Comment:
Maybe a note about `-g` as well?

Reply:
I'm not following, can you clarify?  (I called these out because they appear on the sample command line, but `-g` doesn't.  Are you saying that you think `-g` should be thrown along with the others and explained?  I haven't tried it, but was imagining it would hurt performance, so not something I'd want to use if profiling to determine where time was being spent/wasted).

Reply:
It's not mentioned already, but its certainly can be useful when running tools like `gprof` or `perf` (and I think the original gitter thread had it in the invocation?). It can get in the way of optimization a little but I personally find it helpful for understanding call stacks

Reply:
Hmm, you're right, the gitter thread does, I hadn't noticed.

I feel sufficiently non-expert here that I'm disinclined to add it since I'm mostly trying to do damage control and not create the profiling section that I think we should have (which should arguably be in user-facing docs rather than developer-oriented docs, have more details about other options, etc.).  I.e., I'd prefer to rely on developers who get this far knowing what else they might throw to make it better.  You can talk me into it, though, if you like.

Arguably, the `--fast` and `--savec` flags should also be eliminated based on my argument, and I probably wouldn't have included them if I'd been writing this from scratch.  But since they were already here and not doing harm, I was trying to make the smaller editorial change.

Reply:
I don't feel that strongly about it. 

> I'd prefer to rely on developers who get this far knowing what else they might throw to make it better.

This logic works for me